src/Symfony/Component/HttpKernel/Controller/TraceableControllerResolver.php

```diff
@@ -19,21 +19,28 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-class TraceableControllerResolver implements ControllerResolverInterface
+class TraceableControllerResolver implements ControllerResolverInterface, ArgumentResolverInterface
 {
     private $resolver;
     private $stopwatch;
+    private $argumentResolver;
 
     /**
      * Constructor.
      *
-     * @param ControllerResolverInterface $resolver  A ControllerResolverInterface instance
-     * @param Stopwatch                   $stopwatch A Stopwatch instance
+     * @param ControllerResolverInterface $resolver         A ControllerResolverInterface instance
+     * @param Stopwatch                   $stopwatch        A Stopwatch instance
+     * @param ArgumentResolverInterface   $argumentResolver Only required for BC
      */
-    public function __construct(ControllerResolverInterface $resolver, Stopwatch $stopwatch)
+    public function __construct(ControllerResolverInterface $resolver, Stopwatch $stopwatch, ArgumentResolverInterface $argumentResolver = null)
     {
         $this->resolver = $resolver;
         $this->stopwatch = $stopwatch;
+        $this->argumentResolver = $argumentResolver;
+
+        if (null === $this->argumentResolver) {
```

Comment:
Should be implied by the getArguments method deprecation but adding it just in case